### PR TITLE
Add httpsServerOptions to ServerOptions interface

### PR DESCRIPTION
**Pull request by toefraz**
http://restify.com/#creating-a-server

Adding missing httpsServerOptions property to ServerOptions interface.

### CODE REVIEW

File: restify/restify.d.ts [hunk 1]
```diff
@@ -452,6 +452,7 @@ declare module "restify" {
     responseTimeFormatter ?: (durationInMilliseconds: number) => any;
     handleUpgrades ?: boolean;
     router ?: Router;
+    httpsServerOptions?: Object;
```
Comment (vvakame) [inline restify/restify.d.ts]: please use `any` instead of `Object`.

Comment (toefraz) [inline restify/restify.d.ts]: Thanks for the input! Could you explain why `any` is preferred over `Object`?

The `httpsServerOptions` property should be undefined or null or only be given an Object as a value. If the property exists, Restify passes the the property into https.createServer [[1]](https://github.com/restify/node-restify/blob/5.x/lib/server.js#L291) which expects an Object [[2]](https://github.com/nodejs/node/blob/master/lib/https.js#L12).

Comment (vvakame) [inline restify/restify.d.ts]: https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/declaration%20files/Do's%20and%20Don'ts.md#number-string-boolean-and-object
> If you're tempted to use the type Object, consider using any instead. There is currently no way in TypeScript to specify an object that is "not a primitive".

Comment (toefraz) [inline restify/restify.d.ts]: Cool. Thanks for the info :+1:

**Pull request merged**